pennylane/transforms/compile.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for the high-level quantum function transform that executes compilation."""
+
+from functools import partial
+
+from pennylane import apply
+from pennylane.tape import get_active_tape
+from pennylane.ops import __all__ as all_ops
+
+from pennylane.transforms import single_tape_transform, qfunc_transform
+from pennylane.transforms.optimization import cancel_inverses, commute_controlled, merge_rotations
+
+
+default_pipeline = [commute_controlled, cancel_inverses, merge_rotations]
+
+
+@qfunc_transform
+def compile(tape, pipeline=None, basis_set=None, num_passes=1):
+    """Compile a circuit by applying a series of transforms to a quantum function.
+
+    The default set of transforms includes (in order):
+
+    - pushing all commuting single-qubit gates as far right as possible
+      (:func:`~.commute_controlled`)
+    - cancellation of adjacent inverse gates
+      (:func:`~.cancel_inverses`)
+    - merging adjacent rotations of the same type
+      (:func:`~.merge_rotations`)
```

Comment:
These still aren't linking 🤔 

Note: you can run
```bash
python -msphinx.ext.intersphinx https://pennylane.readthedocs.io/en/latest/objects.inv | grep me
rge_rotations
```
to download the reference names. I gave it a go, looks like `~pennylane.transforms.merge_rotations` might work?